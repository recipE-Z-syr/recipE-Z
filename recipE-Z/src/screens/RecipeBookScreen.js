import React, { Component } from 'react';
import { Button, StyleSheet, Text, TextInput, View, SafeAreaView, ScrollView, Keyboard, TouchableOpacity } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import defaultStyles from './stylesheet';
import NavigationBar from './NavigationBar';

class RecipeBookScreen extends React.Component {
  render() {
    const {navigation} = this.props;
    return (
      <View style={[defaultStyles.container, {backgroundColor:'#ffcece'}]}>
      <NavigationBar navigation={this.props.navigation}/>
      <Button title="Recipe" onPress={() => navigation.navigate('Recipe')} />
      </View>
   ); //end return

 }//end render
}//end class

export default RecipeBookScreen;

const styles = StyleSheet.create({
});
