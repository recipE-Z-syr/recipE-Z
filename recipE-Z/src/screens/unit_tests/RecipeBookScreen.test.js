//RecipeBookScreen.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import RecipeBookScreen from '../RecipeBookScreen';

describe('<RecipeBookScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RecipeBookScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('No account navigates to Sign Up', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<RecipeBookScreen {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(0)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Recipe')
})