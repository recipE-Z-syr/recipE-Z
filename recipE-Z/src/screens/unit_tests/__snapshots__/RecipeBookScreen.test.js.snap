// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeBookScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "fontFamily": "nunito-regular",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#ffcece",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row-reverse",
        "position": "absolute",
        "right": 30,
        "top": 30,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Recipe
    </Text>
  </View>
</View>
`;
